src/States.jl

```diff
@@ -1,10 +1,13 @@
 # Contains the structs that are evolved by the river model, and methods to instantiate them
-export RiverState
+export RiverState, HillslopeChannelRiverState
 
-struct RiverState{AM <: AbstractMatrix}
-    hillslope::AM # instantaneous
-    channel::AM # instantaneous
-end
+abstract type RiverState end
 
-# initialize method
-# initialize_state(river_model::HCRM) where {HCRM <: HillslopChannelRiverModel} end
+struct HillslopeChannelRiverState <: RiverState
+    "Hillslope state history [Dict(id -> Matrix: time-within-lag (end=current) x basin)]"
+    hillslope_state::Dict
+    "Channel state [Dict(id -> Matrix: 1 x basin)]"
+    channel_state::Dict
+    "Date window [DateWindow] of the state history"
```

Comment:
Im not sure if this is being used anywhere. We for sure use the `initial_date_window` made in the example scripts, though

Reply:
This date window is not really used in the code, but i found it very useful when outputting the river state, to have the date-range it was computed for stored nearby